src/hooks/bookmarks/use-bookmarks-by-tutorial-ids.ts

```diff
@@ -0,0 +1,43 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import { Tutorial } from 'lib/learn-client/types'
+import { ApiBookmark } from 'lib/learn-client/api/api-types'
+import { getBookmarksByTutorialIds } from 'lib/learn-client/api/bookmark'
+import useAuthentication from 'hooks/use-authentication'
+
+interface UseBookmarksByTutorialIdsOptions {
+	tutorialIds: Tutorial['id'][]
+}
+
+interface UseBookmarksByTutorialIdsResult extends Omit<UseQueryResult, 'data'> {
+	bookmarks: undefined | ApiBookmark[]
+}
+
+/**
+ * Handles fetching and storing bookmarks with React Query for the given
+ * `tutorialIds`.
+ */
+const useBookmarksByTutorialIds = ({
+	tutorialIds,
+}: UseBookmarksByTutorialIdsOptions): UseBookmarksByTutorialIdsResult => {
+	// Get the current user's access token
+	const { session } = useAuthentication()
+	const accessToken = session?.accessToken
+
+	// Fetch all bookmarks with the access token
+	const { data: bookmarks, ...restQueryResult } = useQuery(
+		['bookmarks'],
```

Comment:
Ah good call, I forgot to come back to this. I paused on thinking about this query key because it's not extremely straightforward. We want to make sure this query gets invalidated on add/remove and I wasn't 100% sure (but will test) if the query you're proposing would get invalidated just based on the `"bookmarks"` portion.